sources/ldap.c

```diff
@@ -418,7 +417,8 @@ od_ldap_server_t *od_ldap_server_pull(od_logger_t *logger, od_rule_t *rule,
 		rc = od_ldap_endpoint_wait(le, timeout);
 
 		if (rc == -1) {
-			od_ldap_endpoint_unlock(le);
+			// od_ldap_endpoint_unlock(le); // do we really need that line?
```

Comment:
It seems like unlock in above code is misplaced. But I'm not sure.

Reply:
but we locking it again on line 425 before the next while-iteration. I think that unlock on line 413 - it's ok